### PR TITLE
Implement sem-semantic-release

### PR DESCRIPTION
**Pull request by mattrym**

### CODE REVIEW

File: sem-semantic-release [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+semantic-release::parse_args() {
+    local PARSING_PLUGINS=1
+    local PARSING_BRANCHES=1
+    local DRY_RUN=1
+
+    local SEMANTIC_RELEASE_BRANCHES=()
+
+    set -f
+
+    while [[ $# -gt 0 ]]; do
```
Comment (shiroyasha) [inline sem-semantic-release:18]: I'm not sure about this, but I know something that can help.
Please add sem-semantic-relase to the Bash linter block in the Semaphore YAML.

File: sem-semantic-release [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            ;;
+
+        --dry-run)
+            DRY_RUN=0;
+            shift
+            ;;
+
+        -*|--*)
+            echo "sem-semantic-release: Unknown option $1"
+            return 1
```
Comment (shiroyasha) [inline sem-semantic-release:58]: This will need to be exit now that this is a script.

File: sem-semantic-release [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+semantic-release::install() {
+    if [[ -n $SEMANTIC_RELEASE_VERSION ]]
+    then 
+        SEMANTIC_RELEASE_PACKAGE="semantic-release@$SEMANTIC_RELEASE_VERSION"
+    else 
+        SEMANTIC_RELEASE_PACKAGE="semantic-release"
+    fi
+    
+    npm install $SEMANTIC_RELEASE_PACKAGE --silent || 
+        { echo "sem-semantic-release: Unsupported semantic-release version: $SEMANTIC_RELEASE_VERSION"; return 1; }
```
Comment (shiroyasha) [inline sem-semantic-release:103]: Check if this needs to be an exit instead of return.

File: sem-semantic-release [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    
+    if [[ ${#SEMANTIC_RELEASE_PLUGINS[@]} -ne 0 ]]
+    then
+        npm install ${SEMANTIC_RELEASE_PLUGINS[@]} --silent || 
+            { echo "sem-semantic-release: Unable to install plugins: $SEMANTIC_RELEASE_PLUGINS"; return 1; }
+    fi
+}
+
+semantic-release::scrape_version() {
+    RELEASE_VERSION=$(cat /tmp/semantic-release.log | grep "The next release version" | grep -oE '([0-9]+\.[0-9]+\.[0-9]+)')
```
Comment (shiroyasha) [inline sem-semantic-release:113]: Should this be local?

File: sem-semantic-release [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    
+    if [[ -n $RELEASE_VERSION ]]
+    then
+        sem-context put RELEASE_VERSION=$RELEASE_VERSION
+        echo "sem-semantic-release: RELEASE_VERSION=$RELEASE_VERSION"
+    else
+        echo "sem-semantic-release: RELEASE_VERSION not found"
```
Comment (shiroyasha) [inline sem-semantic-release:120]: Change this to:

```
sem-semantic-release: RELEASE_VERSION not found. Not generating a new release.
```

File: sem-semantic-release [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    fi
+}
+
+sem-semantic-release() {
+    semantic-release::parse_args $@
+    semantic-release::install
+    
+    { npx semantic-release $SEMANTIC_RELEASE_OPTIONS; SEMANTIC_RELEASE_RESULT=$?; } | tee /tmp/semantic-release.log 
+    if [[ $SEMANTIC_RELEASE_RESULT -ne 0 ]]; then return $SEMANTIC_RELEASE_RESULT; fi
```
Comment (shiroyasha) [inline sem-semantic-release:129]: Check if this needs to be an exit.

File: sem-semantic-release (hunk 5 above)
Comment (mattrym) [reply sem-semantic-release]: I've put "New release hasn't been generated", because it runs _post factum_.

**Review by shiroyasha — APPROVED** [review]
Looks good. Merge it 🚀

**Pull request merged**